### PR TITLE
Get drone center and polygon

### PR DESCRIPTION
**Pull request by dafnaweiss**

### CODE REVIEW

File: api/controllers/main.js [hunk 1]
```diff
@@ -16,8 +16,9 @@ const fetchDataFromCesium = (async (requestParams) => {
 			return droneCesium ? droneCesium.fetchViewDataRequest(params) : Promise.reject('Cant select "drone-cesium"');
 		}, requestParams);
 	} catch (error) {
+		console.log('error', error);
 		data = error;
-		throw new Error(error);
+		// throw new Error(error);
```
Comment (yairtawil) [inline api/controllers/main.js]: recomment,
remove data

File: api/swagger/geometries.json [hunk 2]
```diff
@@ -1,5 +1,11 @@
 {
   "definitions": {
+    "FeatureArray": {
+      "type": "array",
```
Comment (yairtawil) [inline api/swagger/geometries.json]: use FeatureArray as definition in featureCollection

File: api/swagger/swagger.json [hunk 3]
```diff
@@ -45,41 +46,45 @@
                 "ExifImageHeight"
               ],
               "properties": {
+                "displayUrl": {
```
Comment (yairtawil) [inline api/swagger/swagger.json]: revert

File: client/src/component/drone-cesium.js [hunk 4]
```diff
@@ -1,18 +1,32 @@
 import Cesium from 'cesium/Cesium';
 import 'cesium/Widgets/widgets.css';
-import { featureCollection, point, polygon } from '@turf/turf';
-
+import {point, polygon} from '@turf/turf';
+
```
Comment (yairtawil) [inline client/src/component/drone-cesium.js]: export examples to another js file

File: client/src/component/drone-cesium.js [hunk 5]
```diff
@@ -104,39 +117,53 @@ export class DroneCesiumComponent extends HTMLElement {
...
+		let centerPoint;
+		let bboxPolygon;
+
+		// try to get the Center Point
+		try {
+			const center = this.viewer.camera.getPickRay({x: canvas.width / 2, y: canvas.height / 2});
+			const centerCoordinates = this.toCartographicDegress(center);
+			centerPoint = point(centerCoordinates);
+		} catch (error) {
+			centerPoint = {};
```
Comment (yairtawil) [inline client/src/component/drone-cesium.js]: null

File: client/src/component/drone-cesium.js [hunk 6]
```diff
@@ -104,39 +117,53 @@ export class DroneCesiumComponent extends HTMLElement {
...
+		try {
+			const topLeft = this.viewer.camera.getPickRay({x: 0, y: 0});
+			const topRight = this.viewer.camera.getPickRay({x: canvas.width, y: 0});
+			const bottomRight = this.viewer.camera.getPickRay({x: canvas.width, y: canvas.height});
+			const bottomLeft = this.viewer.camera.getPickRay({x: 0, y: canvas.height});
+
+			const bboxCoordinates = [topLeft, topRight, bottomRight, bottomLeft, topLeft].map(this.toCartographicDegress.bind(this));
+			bboxPolygon = polygon([bboxCoordinates]);
+		} catch (error) {
+			bboxPolygon = {};
```
Comment (yairtawil) [inline client/src/component/drone-cesium.js]: null

File: client/src/component/drone-cesium.js [hunk 7]
```diff
@@ -104,39 +117,53 @@ export class DroneCesiumComponent extends HTMLElement {
...
+		}
+		return [bboxPolygon, centerPoint];
```
Comment (yairtawil) [inline client/src/component/drone-cesium.js]: return an object:

```javascript
{
   centerPoint,
   bboxPolygon
}
```

define inside swagger definitions

File: client/examples/JSONexamples.json [hunk 8]
```diff
@@ -0,0 +1,24 @@
+{
```
Comment (yairtawil) [inline client/examples/JSONexamples.json]: remove

File: client/src/component/drone-cesium.js [hunk 9]
```diff
@@ -107,112 +99,84 @@ export class DroneCesiumComponent extends HTMLElement {
...
-		const centerCoordinates = this.toCartographicDegress(center);
-		const bboxPolygon = polygon([bboxCoordinates]);
-		const centerPoint = point(centerCoordinates);
-		return featureCollection([bboxPolygon, centerPoint]);
+		const scene = this.viewer.scene;
+		let centerPoint;
+		let bboxPolygon;
+
+		// try to get the Center Point
+		try {
```
Comment (yairtawil) [inline client/src/component/drone-cesium.js]: remove all console logs

File: client/src/component/drone-cesium.js [hunk 10]
```diff
@@ -107,112 +99,84 @@ export class DroneCesiumComponent extends HTMLElement {
...
-		const cartesian = this.viewer.scene.globe.pick(position, this.viewer.scene);
-		const cartographic = Cesium.Cartographic.fromCartesian(cartesian);
-
-		const longitude = Cesium.Math.toDegrees(cartographic.longitude);
-		const latitude = Cesium.Math.toDegrees(cartographic.latitude);
-		return [longitude, latitude];
+	toCartographicDegress(position, scene) {
+		console.log(`start toCartographicDegress...position: ${JSON.stringify(position)}`);
+		const cartesian = this.viewer.scene.globe.pick(position, scene);
+		console.log(`cartesian: ${cartesian}`);
```
Comment (yairtawil) [inline client/src/component/drone-cesium.js]: remove

File: client/src/component/drone-cesium.js [hunk 11]
```diff
@@ -107,112 +99,84 @@ export class DroneCesiumComponent extends HTMLElement {
...
+			const latitude = Cesium.Math.toDegrees(cartographic.latitude);
+			console.log(`toCartographicDegress longitude: ${longitude}`);
+			console.log(`toCartographicDegress longitude: ${latitude}`);
+			return [longitude, latitude];
+		} else {
+			return [];
+		}
 	}
-
-	// viewer;
```
Comment (yairtawil) [inline client/src/component/drone-cesium.js]: good!

File: client/src/component/drone-cesium.js (hunk 9 above)
Comment (dafnaweiss) [reply client/src/component/drone-cesium.js]: I left only the points's coordinates console-log. till we found another solution to the timeout...

**Pull request merged**